packages/fastify/modules/plugin/plugin.js

```diff
@@ -0,0 +1,81 @@
+const fp = require('fastify-plugin');
+const merge = require('merge-options');
+const {
+  Prometheus,
+  createMetricTypes,
+  createRequestRecorder,
+  createGcObserver,
+  defaultNormalizers,
+  getSummary,
+  getContentType,
+} = require('@promster/metrics');
+
+const defaultRouteOptions = {
+  route: { method: 'GET', url: '/metrics' },
+};
+
+let recordRequest;
+let upMetric;
+
+const extractPath = req => req.raw.originalUrl || req.raw.url;
+const getRequestRecorder = () => recordRequest;
+const signalIsUp = () => upMetric && upMetric.set(1);
+const signalIsNotUp = () => upMetric && upMetric.set(0);
+
+const plugin = async function(fastify, options) {
+  const defaultedOptions = merge(
+    createMetricTypes.defaultedOptions,
+    createRequestRecorder.defaultedOptions,
+    defaultNormalizers,
+    defaultRouteOptions,
+    options
+  );
+
+  const metricTypes = createMetricTypes(defaultedOptions);
+  const observeGc = createGcObserver(metricTypes);
+
+  recordRequest = createRequestRecorder(metricTypes, defaultedOptions);
+  upMetric = metricTypes && metricTypes.up;
+
+  observeGc();
+
+  fastify.decorate('Prometheus', Prometheus);
+  fastify.decorate('recordRequest', recordRequest);
+  fastify.decorateRequest('__promsterStartTime__', 0);
```

Comment:
I don't think this will give accurate metrics. This should also be hrtime I think. Otherwise the request time will be far too long.

Reply:
this is a default value which is overwritten for every request inside `onRequest` handler
so whatever we pass here, will be wrong in the context of the request and it won't affect anything.
What about removing the second argument at all to avoid confusion?
In this case `__promsterStartTime__` will be `undefined` before `onRequest` initialization

Reply:
I like that yeah. Thanks for the explanation.